tests/test_filters.py

```diff
@@ -52,6 +57,57 @@ def test_flate_decode_encode(predictor, s):
     assert codec.decode(encoded, DictionaryObject({"/Predictor": predictor})) == s
 
 
+@pytest.mark.parametrize("s", filter_inputs)
+def test_brotli_decode_encode(s):
+    """BrotliDecode encode() and decode() methods work as expected."""
+    codec = BrotliDecode()
+    s_bytes = s.encode()
+    encoded = codec.encode(s_bytes)
+    assert encoded != s_bytes  # Ensure encoding actually happened
+    decoded = codec.decode(encoded)
+    assert decoded == s_bytes
+
+
+
+@patch("pypdf.filters.brotli", None)
+def test_brotli_missing_installation_mocked():
+    """Verify BrotliDecode raises ImportError if brotli is not installed."""
+    from pypdf.filters import BrotliDecode, decode_stream_data
```

Comment:
Why do we need to have this local?

Reply:
Could you please answer or resolve this question?

Reply:
@ash01ish Could you please answer here? This currently prevents me from further reviewing your changes.